thoth/storages/graph/postgres.py

```diff
@@ -489,67 +489,835 @@ def has_python_solver_error(
+        python_version: str = None
+    ) -> Query:
+        """Construct query for unsolved Python packages versions functions, the query is not executed."""
+        solved = self._construct_solved_query(
+            package_name=package_name,
+            package_version=package_version,
+            index_url=index_url,
+            os_name=os_name,
+            os_version=os_version,
+            python_version=python_version
+        )
+
+        subquery = solved.subquery()
+        query = (
+            self._session.query(PythonPackageVersionEntity)
+            .join(PythonPackageIndex)
+            .filter(
+                tuple_(
+                    PythonPackageVersionEntity.package_name,
+                    PythonPackageVersionEntity.package_version)
+                .notin_(
+                    subquery
+                )
+            )
+            .with_entities(
+                PythonPackageVersionEntity.package_name,
+                PythonPackageVersionEntity.package_version,
+                PythonPackageIndex.url)
         )
 
         return query
 
-    def retrieve_unsolved_python_packages(
+    def get_unsolved_python_package_versions(
         self,
-        solver_name: str = None,
-        randomize: bool = True
-    ) -> List[Tuple[str, str]]:
-        """Retrieve a list of tuples (package name, package version) of dependencies which were not yet resolved.
+        *,
+        start_offset: int = 0,
+        count: int = _DEFAULT_COUNT,
+        package_name: str = None,
+        package_version: str = None,
+        index_url: str = None,
+        os_name: str = None,
+        os_version: str = None,
+        python_version: str = None,
+        distinct: bool = False,
+    ) -> List[Tuple[str, str, str]]:
+        """Retrieve unsolved Python package versions in Thoth Database.
+
+        Examples:
+        >>> from thoth.storages import GraphDatabase
+        >>> graph = GraphDatabase()
+        >>> graph.get_python_package_versions()
+        [('regex', '2018.11.7', 'https://pypi.org/simple'), ('tensorflow', '1.11.0', 'https://pypi.org/simple')]
+        """
+        query = self._construct_unsolved_python_package_versions_query(
+            package_name=package_name,
+            package_version=package_version,
+            index_url=index_url,
+            os_name=os_name,
+            os_version=os_version,
+            python_version=python_version
+            )
+
+        if index_url is not None:
+            query = query.filter(PythonPackageIndex.url == index_url)
+
+        if package_name is not None:
+            query = query.filter(PythonPackageVersionEntity.package_name == package_name)
+
+        if package_version is not None:
+            query = query.filter(PythonPackageVersionEntity.package_version == package_version)
+
+        query = query.offset(start_offset).limit(count/2)
+
+        if distinct:
+            query = query.distinct()
+
+        query = query.all()
```

Comment:
Just rename `query` to result here:

```
        result = query.all()
```

As this code gets result, the return might be misleading.